core/src/test/scala/org/apache/spark/scheduler/MapStatusSuite.scala

```diff
@@ -128,4 +130,23 @@ class MapStatusSuite extends SparkFunSuite {
     assert(size1 === size2)
     assert(!success)
   }
+
+  test("Blocks which are bigger than 2 * average size should not be underestimated.") {
+    val sizes = Array.concat(Array.fill[Long](1000)(1L), (1000L to 2000L).toArray)
+    val status1 = MapStatus(BlockManagerId("exec-0", "host-0", 100), sizes)
+    val arrayStream = new ByteArrayOutputStream(102400)
+    val objectOutputStream = new ObjectOutputStream(arrayStream)
+    assert(status1.isInstanceOf[HighlyCompressedMapStatus])
+    status1.asInstanceOf[HighlyCompressedMapStatus].writeExternal(objectOutputStream)
+    objectOutputStream.flush()
+    val array = arrayStream.toByteArray
+    val objectInput = new ObjectInputStream(new ByteArrayInputStream(array))
+    val status2 = new HighlyCompressedMapStatus()
+    status2.readExternal(objectInput)
```

Comment:
I'm a little bit confused and hesitant here.
We `writeExternal` for serialization and do initialization in `readExternal` when deserialize, right?
`objectInput.readObject().asInstanceOf[HighlyCompressedMapStatus]` is not ok, I think?